src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -68,6 +68,20 @@ public override BoundNode VisitConversion(BoundConversion node)
 
             var rewrittenType = VisitType(node.Type);
 
+            // special handling for initializers converted to ROS<T>
+            if (node.Operand is BoundConvertedStackAllocExpression stackAllocExpression &&
+                TypeSymbol.Equals(rewrittenType.OriginalDefinition, _compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T), TypeCompareKind.ConsiderEverything) &&
+                Binder.GetWellKnownTypeMember(_compilation, WellKnownMember.System_Runtime_CompilerServices_RuntimeHelpers__CreateSpanRuntimeFieldHandle, _diagnostics, syntax: node.Operand.Syntax, isOptional: true) is not null)
+            {
+                var count = VisitExpression(stackAllocExpression.Count);
+                var elementType = VisitType(stackAllocExpression.ElementType);
+                var initializer = (BoundArrayInitialization?)VisitExpression(stackAllocExpression.InitializerOpt);
+                if (CodeGen.CodeGenerator.UseCreateSpanForReadOnlySpanStackAlloc(elementType, initializer, isEncDelta: this.EmitModule?.IsEncDelta == true))
```

Comment:
>if (CodeGen.CodeGenerator.UseCreateSpanForReadOnlySpanStackAlloc(elementType, initializer, isEncDelta: this.EmitModule?.IsEncDelta == true)) [](http://example.com/codeflow?start=16&length=140)

Are we visiting things twice if this condition isn't satisfied? That would be a wrong thing to do.